regularizepsf/visualize.py

```diff
@@ -0,0 +1,125 @@
+        corners.append(corner)
+        counts.append(count)
+    # corners has the lower-left corner of each patch
+    corners = np.array(corners)
+    corners += patch_collection.size // 2
+    # Now it has the center of each patch
+
+    ax.set_aspect('equal')
+    m = ax.scatter(corners[:, 1], corners[:, 0], c=counts, s=200)
+    plt.colorbar(m).set_label(
+            "Number of stars found in patch centered on point")
+
+    return ax
+
+
+def visualize_patches(patch_collection: PatchCollectionABC,
+                      ax: matplotlib.axes.Axes = None,
+                      psf_size: int = None,
+                      figsize: tuple[float, float] = None,
+                      uniform_scaling: bool = False,
+                      imshow_args: dict = {}) -> matplotlib.axes.Axes:
+    """
+    Utility to visualize the patches in a PatchCollection
+
+    Parameters
+    ----------
+    patch_collection : PatchCollectionABC
+        A patch collection, such as that returned by
+        `CoordinatePatchCollection.find_stars_and_average`.
+    ax : matplotlib.axes.Axes
+        An Axes object on which to plot. If not provided, a new Figure will be
+        generated.
+    psf_size : int
+        If the PatchCollection was generated using a PSF size smaller than the
+        patch size, provide the psf_size here to trim the padding from each
+        patch.
+    figsize : tuple
+        If `ax` is not provided, the size of the generated Figure can be set.
+    uniform_scaling : boolean
+        If True, use the scame colormap scaling for all patches. If False,
+        normalize each one separately.
+    imshow_args : dict
+        Additional arguments to pass to each `plt.imshow()` call
+    """
+    if not patch_collection.patches:
+        raise ValueError("This PatchCollection does not have any patches")
+
+    if ax is None:
+        fig = plt.figure(figsize=figsize)
+        ax = fig.subplots()
+
+    patch_size = patch_collection.size
+    if psf_size is not None:
+        trim = int((patch_size - psf_size) / 2)
+
+    if uniform_scaling:
+        # Work out vmin/vmax values that work for all patches
+        vmin, vmax = np.inf, -np.inf
+        for patch in patch_collection.values():
+            if psf_size is not None and trim:
+                patch = patch[trim:-trim, trim:-trim]
+            vmin = min(vmin, patch.min())
+            vmax = max(vmax, patch.max())
+    else:
+        vmin, vmax = None, None
+
+    kwargs = {'vmin': vmin, 'vmax': vmax}
+    kwargs.update(imshow_args)
+
+    # Track the coordinates we see, to later set the overall plot bounds
+    xs, ys = [], []
+
+    for corner, patch in patch_collection.items():
+        if psf_size is not None and trim:
+            patch = patch[trim:-trim, trim:-trim]
+        # Since the patches overlap each other, we need to plot each one so it
+        # only reaches out to the point of overlap.
+        extent = (corner.y + patch_size/4,
```

Comment:
I think it should be possible both to plot a fraction of the patches, and to plot all the patches. The former may be good for cleaner illustrations, while the latter is good for checking that all the patches look reasonable.

One thing I was unsure about in my code was plotting all the patches (from all four grids) together inside one large set of axes labeled with pixel numbers. The axis labels help connect a patch to an image location, but they misrepresent the range of each patch, since I was cramming them all into one big plot despite their sample overlapping regions. I like that your approach removes all the axis labels and adds small borders between each patch, to make their locations more representative than pixel-perfect.